modules/core/04-channel/keeper/packet.go

```diff
@@ -462,6 +470,10 @@ func (k Keeper) AcknowledgePacket(
 
 	commitment := k.GetPacketCommitment(ctx, packet.GetSourcePort(), packet.GetSourceChannel(), packet.GetSequence())
 
+	if len(commitment) == 0 {
+		return sdkerrors.Wrapf(types.ErrPacketCommitmentNotFound, "packet with sequence (%d) has been acknowledged, or timed out. In rare cases the packet was never sent or the packet sequence is incorrect", packet.GetSequence())
```

Comment:
What do you mean, "in rare cases"? In the case of incorrect relayer behaviour, perhaps?

Reply:
correct, should I rephrase? My thought process is it seems very unlikely a relayer would have a proof for a packet never sent (I'm not even sure how this would happen). The only case I can think of is if the relayer sets the packet sequence incorrectly

Reply:
I think it's worth noting that this indicates some relayer misconfiguration, yeah. Very minor though.